docs/variant.md

```diff
@@ -1,207 +1,200 @@
-  };
-```
-
-### Honorable Mentions
-
-The [standard library](/api/index.html) exposes two important variants you'll come to hear a lot.
+## Usage
 
-#### `option`
+The primary way to interact with variants is through
+[pattern matching](pattern-matching.md). When pattern matching over a variant
+the compiler can ensure that all cases are covered exhaustively so that there
+is no undefined behavior. This is also helpful when adding new cases to a
+variant later, the compiler will present a clear list of code that needs
+to be updated.
 
 ```reason
-type option('a) = None | Some('a);
+let interact = (animal) => {
+  switch (animal) {
+  | Cat => "pet"
+  | Dog => "throw ball"
+  | Horse => "ride"
+  | Snake => "run away"
+  };
+};
 ```
 
-This is the convention used to simulate a "nullable" (aka `undefined` or `null`) value in other languages. Thanks to this convenience type definition, Reason can default every value to be non-nullable. An `int` will always be an int, never "`int` **or** `null` **or** `undefined`". If you do want to express a "nullable int", you'd use `option(int)`, whose possible values are `None` or `Some(int)`. `switch` forces you to handle both cases; therefore, **a pure Reason program doesn't have null errors**.
-
-#### `list`
+It is possible to interact with a constructor's data while pattern matching:
 
 ```reason
-type list('a) = Empty | Head('a, list('a));
+let checkFirst = (linkedList) => {
+  switch (linkedList) {
+  | Node(value, _) => "First is:" ++ string_of_int(value)
+  | Empty => "The list is empty"
+  };
+};
 ```
 
-_Not the actual type definition. Just an illustration_.
+## Option
 
-This says: "a list that holds a value of type `a` (whatever it is) is either empty, or holds that value plus another list".
+Options are a commonly used variant built into the Reason language. They
+represent the presence or absence of a value. It is similar to the concept of
+"nullable" values in other languages.
 
-Reason gave `list` a syntax sugar. `[1, 2, 3]` is conceptually equivalent to `Head(1, Head(2, Head(3, Empty)))`. Once again, `switch` forces you to handle every case of this variant, including `Empty` (aka `[]`). **This eliminates another big category of access bugs**.
+Details: [Options](option.md)
 
-#### Other Variant-like Types
+## Inline Records
 
-Did you know that you can use `switch` on string, int, float, array, and most other data structures? Try it!
-
-## Tips & Tricks
-
-**Be careful** not to confuse a constructor carrying 2 arguments with a constructor carrying a single tuple argument:
+Defining variant constructors with many arguments is an anti-pattern, just like
+defining a function with many unnamed arguments is an anti-pattern. To give
+the arguments names a record is used:
 
 ```reason
-type account =
-  | Facebook(string, int) /* 2 arguments */;
-type account2 =
-  | Instagram((string, int)) /* 1 argument - happens to be a 2-tuple */;
-```
-### Variants Must Have constructors
+type cat = {
```

Comment:
Might make sense to actually include more than one field in this since the docs just stated a motivation to avoid "many arguments" (weird to see only one here).

Reply:
Or you could just omit the first paragraph as people already know this intuitively, and instead setup the example. Often times variants will hold a single predefined type..
`<example>`
But if that record type is only ever used within a single variant, it is often more concise to define the variant and the record type at the same time.

Reply:
Good point, I'll rephrase the setup